homeassistant/components/homewizard/number.py

```diff
@@ -0,0 +1,69 @@
+"""Creates HomeWizard Number entities."""
+from __future__ import annotations
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import HWEnergyDeviceUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up numbers for device."""
+    coordinator: HWEnergyDeviceUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    if coordinator.data["state"]:
+        async_add_entities(
+            [
+                HWEnergyNumberEntity(coordinator, entry),
+            ]
+        )
+
+
+class HWEnergyNumberEntity(
+    CoordinatorEntity[HWEnergyDeviceUpdateCoordinator], NumberEntity
+):
+    """Representation switchable entity."""
+
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: HWEnergyDeviceUpdateCoordinator,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize the switch."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{entry.unique_id}_status_light_level"
+        self._attr_name = "Light level"
+        self._attr_native_unit_of_measurement = PERCENTAGE
+        self._attr_icon = "mdi:lightbulb-on"
```

Comment:
So... why is this a `number` and not ` light`?

Reply:
I have considered this, but this is only a control for when the socket is 'on'. And you can only control its brightness, not the on/off state. Also, it is a _status_ light, not a light.

For example, if it was a light and you turn if off, then you only set the brightness to `0`, if you turn it back on, what should be the value then?

It feels more logic to me to make it a configurable number. But of course I am open to other opinions 🙂.

Reply:
If you can't control the status light on/off, it cannot be a `light`, so makes sense. Thanks 👍